fast_llm/data/data/gpt/data.py

```diff
@@ -77,18 +79,43 @@ def __init__(
         self._max_sequence_length = max_sequence_length
         self._cross_document_attention = cross_document_attention
 
+        # This is a mapping between the dataset name in the config and its normalized variant for later reference.
+        # This is needed because previously, dataset names were based on phases and were capitalized,
+        # but this is no longer a requirement.
+        self._dataset_name_map = {key.lower(): key for key in self._config.datasets.keys()}
```

Comment:
How about we only rename known values (`Training`, `Validation`, `Test`) in `_from_dict` instead? 
(Also redundant with the `__post_init__` thing)